homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,254 @@
+        devices.append(setup_hvac(ip_address, name, customize))
+
+    add_devices(devices)
+
+
+def setup_hvac(ip_address, name, customize):
+    if ip_address is None:
+        _LOGGER.error("Missing required configuration items %s",
+                      CONF_IP_ADDRESS)
+        return False
+
+    import pydaikin.appliance as appliance
+    device = appliance.Appliance(ip_address)
+
+    if name is None:
+        name = device.values['name']
+
+    return DaikinHVAC(device, name, customize)
+
+
+class DaikinHVAC(ClimateDevice):
+    """Representation of a Daikin HVAC."""
+
+    def __init__(self, device, name, customize):
+        import pydaikin.appliance as appliance
+
+        """Initialize the HVAC."""
+        self._name = name
+        self._device = device
+        self._customize = customize
+
+        self._current_temperature = self.settings('htemp', True)
+        self._operation_list = list(DAIKIN_OPERATION_LIST.values())
+        self._fan_list = appliance.daikin_values('f_rate')
+        self._swing_list = appliance.daikin_values('f_dir')
+
+        self._target_temperature = None
+        self._current_humidity = None
+        self._target_humidity = None
+        self._current_fan_mode = None
+        self._current_swing_mode = None
+
+    @property
+    def unique_id(self):
+        """Return the ID of this AC."""
+        return "{}.{}".format(self.__class__, self._device.ip)
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def should_poll(self):
+        """Polling needed for a demo climate device."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        self._current_temperature = self.settings('htemp', True)
+
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        self._target_temperature = self.settings('stemp', True)
+
+        return self._target_temperature
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+
+        if temperature is None:
+            return
+
+        self._device.set({"temp": temperature})
+        self._target_temperature = temperature
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        state = self._device.represent('mode')[1]
+
+        current_operation = DAIKIN_STATE_TO_HA.get(state)
+        if current_operation in [STATE_ECO, STATE_PERFORMANCE]:
```

Comment:
Why don't just just return `STATE_ECO` or `STATE_PERFORMANCE`?

Reply:
Because I prefer to return always an answer from the all functions in a variable in a single point and the return statement is anyway on the next line. Why have 2 returns one under another ?

And in the code above Daikin reports a status which should be translated in HA as an STATE_AUTO.
I'm basically overriding that return answer for HA

Reply:
The latter was what I was wondering about. Why should `STATE_ECO` be translated to `STATE_AUTO`? Why not tell the user that the device is currently in `STATE_ECO`?

Reply:
Because there's no ECO or PERFORMANCE mode it's only the AUTO button. Take it as an internal mode on Daikin AC